src/Core/src/Platform/iOS/KeyboardAutoManagerScroll.cs

```diff
@@ -0,0 +1,582 @@
+using CoreGraphics;
+using Foundation;
+using UIKit;
+
+namespace Microsoft.Maui.Platform;
+
+internal static class KeyboardAutoManagerScroll
+{
+	static UIScrollView? LastScrollView;
+	static CGPoint StartingContentOffset;
+	static UIEdgeInsets StartingScrollIndicatorInsets;
+	static UIEdgeInsets StartingContentInsets;
+	static CGRect KeyboardFrame = CGRect.Empty;
+	static CGPoint TopViewBeginOrigin = new(nfloat.MaxValue, nfloat.MaxValue);
+	static readonly CGPoint InvalidPoint = new(nfloat.MaxValue, nfloat.MaxValue);
+	static double AnimationDuration = 0.25;
+	static UIView? View = null;
+	static UIView? ContainerView = null;
+	static CGRect? CursorRect = null;
+	internal static bool IsKeyboardShowing = false;
+	static int TextViewTopDistance = 20;
+	static int DebounceCount = 0;
+	static NSObject? WillShowToken = null;
+	static NSObject? WillHideToken = null;
+	static NSObject? TextFieldToken = null;
+	static NSObject? TextViewToken = null;
+
+	internal static void Init()
+	{
+		TextFieldToken = NSNotificationCenter.DefaultCenter.AddObserver(new NSString("UITextFieldTextDidBeginEditingNotification"), DidUITextBeginEditing);
+
+		TextViewToken = NSNotificationCenter.DefaultCenter.AddObserver(new NSString("UITextViewTextDidBeginEditingNotification"), DidUITextBeginEditing);
+
+		WillShowToken = NSNotificationCenter.DefaultCenter.AddObserver(new NSString("UIKeyboardWillShowNotification"), WillKeyboardShow);
+
+		WillHideToken = NSNotificationCenter.DefaultCenter.AddObserver(new NSString("UIKeyboardWillHideNotification"), WillHideKeyboard);
+	}
+
+	static async void DidUITextBeginEditing(NSNotification notification)
+	{
+		if (notification.Object is not null)
+		{
+			View = (UIView)notification.Object;
+
+			if (View is null)
+				return;
+
+			CursorRect = null;
+
+			ContainerView = View.SetContainerView();
+
+			// the cursor needs a small amount of time to update the position
+			await Task.Delay(5);
+
+			var localCursor = FindLocalCursorPosition();
+			if (localCursor is CGRect local)
+				CursorRect = View.ConvertRectToView(local, null);
+
+			TextViewTopDistance = localCursor is CGRect cGRect ? 20 + (int)cGRect.Height : 20;
+
+			await AdjustPositionDebounce();
+		}
+	}
+
+	static CGRect? FindLocalCursorPosition()
+	{
+		if (View is IUITextInput textInput)
+		{
+			var selectedTextRange = textInput.SelectedTextRange;
+			if (selectedTextRange is UITextRange selectedRange)
+				return textInput.GetCaretRectForPosition(selectedRange.Start);
+		}
+
+		return null;
+	}
+
+	internal static CGRect? FindCursorPosition()
+	{
+		var localCursor = FindLocalCursorPosition();
+		if (localCursor is CGRect local)
+		{
+			return View?.ConvertRectToView(local, null);
+		}
+
+		return null;
+	}
+
+	static async void WillKeyboardShow(NSNotification notification)
+	{
+		var userInfo = notification.UserInfo;
+
+		if (userInfo is not null)
+		{
+			if (userInfo.TryGetValue(new NSString("UIKeyboardFrameEndUserInfoKey"), out var frameSize))
```

Comment:
NSString is a IDisposable class, which  is not being disposed here. Can we have this as a static object?